src/components/RangeSlider/RangeSlider.tsx

```diff
@@ -366,48 +348,53 @@ export const RangeSlider = ({
       });
 
       if (closestVal) {
-        // TODO: use the closest val to find the handle to move and move it
-        handleDrag(clickedValue);
+        if (onDrag) onDrag(clickedValue);
+        if (onChange) onChange(clickedValue);
+        if (onDebounceChange) onDebounceChange(clickedValue);
+        if (onRelease) onRelease(clickedValue);
+
         if (slideRailProps.onMouseDown && typeof slideRailProps.onMouseDown === 'function') {
           e.persist();
           slideRailProps.onMouseDown(e);
         }
       }
     },
-    [slideRailProps, sliderBounds, handleDrag, domain, processedValues],
+    [
+      slideRailProps,
+      sliderBounds,
+      onChange,
+      onDebounceChange,
+      onRelease,
+      onDrag,
+      domain,
+      processedValues,
+    ],
   );
-  const handleSlideRailClickWithAnalytics = (e: any) =>
+  const handleSlideRailClickWithAnalytics = (e: MouseEvent) =>
     handleEventWithAnalytics('RangeSlider', handleSlideRailClick, 'onClick', e, containerProps);
 
   const bind = useDrag(
-    ({ active, down, movement: [deltaX, deltaY], vxvy: [vx] }) => {
+    ({ down, movement: [deltaX] }) => {
       const delta = (deltaX / sliderBounds.width) * domain;
       valueBuffer.current = clamp(delta, min, max);
-      if (motionBlur) {
-        requestAnimationFrame(() => {
-          const blurSize = Math.round(Math.abs(vx * 10)) || 0;
-          if (blurRef.current === null) {
-            return;
-          }
-          // eslint-disable-next-line @typescript-eslint/ban-ts-comment
-          // @ts-ignore ts(2531)
-          blurRef.current.setAttribute('stdDeviation', `${down && active ? blurSize : 0}, 0`);
-        });
-      }
 
       setDraggedHandle(down ? 0 : -1);
-      debouncedDrag();
-      set({
-        x: down ? deltaX : pixelPositions[0],
-        y: down ? deltaY : 0,
+      handleDrag(valueBuffer.current);
 
-        immediate: prefersReducedMotion || springOnRelease ? down : true,
+      let animate = true;
+      if (prefersReducedMotion) animate = false;
+      if (!snapToValue) animate = springOnRelease ? !down : false;
```

Comment:
Trying to simplify your logic here, probably not right but you get the idea
```suggestion
      let animate = !down;
      animate = prefersReducedMotion || !springOnRelease || snapToValue ? false : animate;
```

Reply:
There is some point in the truth table which this doesn't match up. I think to accomplish above we'd need nested ternaries which causes a warning and prob isn't worth it